modules/core/04-channel/v2/types/commitment.go

```diff
@@ -53,13 +55,21 @@ func hashPayload(data Payload) []byte {
 // CommitAcknowledgement returns the V2 acknowledgement commitment bytes. The commitment consists of:
 // sha256_hash(0x02 + sha256_hash(ack1) + sha256_hash(ack2) + ...) from a given acknowledgement.
 func CommitAcknowledgement(acknowledgement Acknowledgement) []byte {
+	// return constant failed acknowledgement commitment if the acknowledgement is unsuccessful
+	// in this case, application modules are expected to only use recv success value
+	// and app acknowledgement will be nil
+	if !acknowledgement.RecvSuccess {
+		return FailedAcknowledgementCommitment[:]
+	}
 	var buf []byte
 	for _, ack := range acknowledgement.GetAppAcknowledgements() {
 		hash := sha256.Sum256(ack)
 		buf = append(buf, hash[:]...)
 	}
 
-	buf = append([]byte{byte(2)}, buf...)
+	// prepend the preimage with the IBC Version and the recv success
+	// to ensure there is no possibility of hash collisions
+	buf = append([]byte{byte(2), byte(1)}, buf...)
```

Comment:
This is a consensus breaking change right? i.e. we need to update this in solidity as well. Would it be possible to avoid this by picking a different `FailedAcknowledgementCommitment`?